homeassistant/components/sensibo/climate.py

```diff
@@ -308,11 +308,13 @@ async def async_set_fan_mode(self, fan_mode: str) -> None:
         if "fanLevel" not in self.device_data.active_features:
             raise HomeAssistantError("Current mode doesn't support setting Fanlevel")
 
+        transformation = self.device_data.fan_modes_to_native
```

Comment:
It looks like the library knows about the Home Assistant integration here.It's just the lowercase names, but the Home Assistant abstraction is leaking into the library here.

I think you could resolve this in one of two ways

- We could have a translation dictionary in the integration to convert to what the lib wants
- The library could accept lowercase or caseful names and we didn't have the transformation in HA as that would avoid implementing Home Assistant specific details in the lib.

Reply:
I see what you mean while I don't necessarily see why it should be a problem. I could obviously rename it to something else than the HA use of words to make it less HA specific but having a dict in the lib to translate lowercase to Sensibo wordings I would not see as a problem?

Option 2 I wanted to avoid as I would have to extend the functions calling the api and do a lot of checking what command the user is sending with it's parameters
Option 1 would mean I would have to typecheck again which I wanted to avoid (by solving the "problem" in the lib instead).

What do you think?

Reply:
If the function is specifically named to translate lower case I think its ok.  Right now its doing the translation for Home Assistant specifically, and ideally the lib knows nothing about Home Assistant.

Reply:
I changed the wording to become less HA specific while the functionality has not changed.
Would prefer to get it in like this.